net/net.go

```diff
@@ -1112,31 +1078,78 @@ func (n *net) startPulling() {
 	}
 }
 
-// getOrCreateOwnLoad returns the first log 'owned' by the host under the given thread.
-// If no log exists, a new one is created under the given thread and returned.
-// This is a strict 'ownership' check vs returning managed logs.
-func (n *net) getOrCreateOwnLog(id thread.ID) (info thread.LogInfo, err error) {
-	thrd, err := n.store.GetThread(id)
+// createLog creates a new log with the given peer as host.
+func (n *net) createLog(id thread.ID, key crypto.Key, identity thread.PubKey) (info thread.LogInfo, err error) {
+	var ok bool
+	if key == nil {
+		info.PrivKey, info.PubKey, err = crypto.GenerateEd25519Key(rand.Reader)
+		if err != nil {
+			return
+		}
+	} else if info.PrivKey, ok = key.(crypto.PrivKey); ok {
+		info.PubKey = info.PrivKey.GetPublic()
+	} else if info.PubKey, ok = key.(crypto.PubKey); !ok {
+		return info, fmt.Errorf("invalid log-key")
+	}
+	info.ID, err = peer.IDFromPublicKey(info.PubKey)
 	if err != nil {
 		return
 	}
-	ownLog := thrd.GetOwnLog()
-	if ownLog != nil {
-		info = *ownLog
+	addr, err := ma.NewMultiaddr("/" + ma.ProtocolWithCode(ma.P_P2P).Name + "/" + n.host.ID().String())
+	if err != nil {
 		return
 	}
-	info, err = createLog(n.host.ID(), nil)
+	info.Addrs = []ma.Multiaddr{addr}
+	// If we're creating the log, we're 'managing' it
+	info.Managed = true
+
+	// Add to thread
+	if err = n.store.AddLog(id, info); err != nil {
+		return info, err
+	}
+	lidb, err := info.ID.MarshalBinary()
 	if err != nil {
-		return
+		return info, err
 	}
-	err = n.store.AddLog(id, info)
-	return
+	if err = n.store.PutBytes(id, identity.String(), lidb); err != nil {
+		return info, err
+	}
+	return info, nil
+}
+
+// getOrCreateLog returns a log for identity under the given thread.
+// If no log exists, a new one is created.
+func (n *net) getOrCreateLog(id thread.ID, identity thread.PubKey) (info thread.LogInfo, err error) {
```

Comment:
This is a fairly significant change: A host can have multiple logs for the same thread that are being authored by different identities. In order to do this, we have to be able to look up a log by an identity. Remember, the first records in a log refers to the author (identity), which allows us to prove to ourselves that it is single-writer.

I'm tracking the relationship between an identity and it's logs in the metadata store.